src/components/Resource/Forms/Training/TrainingSubmit.tsx

```diff
@@ -391,10 +401,24 @@ export const TrainingFormSubmit = ({ model }: { model: NonNullable<TrainingModel
     //   });
     // }
 
-    form.setValue('targetSteps', newSteps);
+    if (form.getValues('targetSteps') !== newSteps) {
+      form.setValue('targetSteps', newSteps);
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [watchFields]);
 
+  useEffect(() => {
+    const eta = calcEta(
+      form.getValues('networkDim'),
+      form.getValues('networkAlpha'),
+      form.getValues('targetSteps'),
+      formBaseModel
+    );
```

Comment:
Can you please check if this works?
```suggestion
    const eta = calcEta(...watchFieldsBuzz, formBaseModel);
```
Another way to do it would be
```suggestion
    const [networkDim, networkAlpha, targetSteps] = watchFieldsBuzz;
    const eta = calcEta(networkDim, networkAlpha, targetSteps, formBaseModel);
```

Reply:
oh wasnt aware that was something that could be done. ill test it.